### PR TITLE
Convert all ports to use `build_port`. NFC

### PR DESCRIPTION
**Pull request by sbc100**
This will allow more easy transition to alternative build systems such as ninja (See #17809).

Also, update build_port to allow separate source and build directories. This avoids copying the entire source code of each port from the extracts archive into the build directory.

### CODE REVIEW

File: tools/ports/freetype.py [hunk 1]
```diff
@@ -77,31 +76,23 @@ def create(final):
...
-                       '-I' + dest_path + '/include',
-                       '-I' + dest_path + '/truetype',
-                       '-I' + dest_path + '/sfnt',
-                       '-I' + dest_path + '/autofit',
-                       '-I' + dest_path + '/smooth',
-                       '-I' + dest_path + '/raster',
-                       '-I' + dest_path + '/psaux',
-                       '-I' + dest_path + '/psnames',
-                       '-I' + dest_path + '/truetype',
-                       '-w',
```
Comment (dschuff) [inline tools/ports/freetype.py:96]: I assume it's intentional that you dropped -w here?

**Review by dschuff — APPROVED** [review]
This looks like a nice simplification although I'm not super-familiar with this code...

File: tools/ports/__init__.py [hunk 2]
```diff
@@ -108,45 +108,39 @@ def install_headers(src_dir, pattern='*.h', target=None):
...
+    if srcs:
+      srcs = [os.path.join(src_dir, s) for s in srcs]
+    else:
+      srcs = []
+      for root, _, files in os.walk(src_dir, topdown=False):
+        if any((excluded in root) for excluded in exclude_dirs):
+          continue
+        for f in files:
+          ext = shared.suffix(f)
+          if ext in ('.c', '.cpp') and not any((excluded in f) for excluded in exclude_files):
```
[the remark this answers is not in the document]
Comment (sbc100) [reply tools/ports/__init__.py:121]: I guess not.  I would imagine we would see failing tests if they did.

File: tools/ports/freetype.py (hunk 1 above)
Comment (sbc100) [reply tools/ports/freetype.py:96]: Yes, build_port takes case or adding `-O2` and `-w`,

**Pull request merged**